server/serializer.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+type TelemetryAPIRequest struct {
+	Event      string
+	Properties map[string]interface{}
+}
+
+func GetTelemetryPayloadFromJSON(data io.Reader) (*TelemetryAPIRequest, error) {
+	body := &TelemetryAPIRequest{}
+	if err := json.NewDecoder(data).Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func IsTelemetryPayloadValid(t *TelemetryAPIRequest) error {
+	if t == nil {
+		return errors.New("invalid request body")
+	}
+
+	if t.Event == "" {
+		return errors.New("event is required")
+	}
+
+	return nil
+}
+
+type AddAPIRequest struct {
+	Message     string `json:"message"`
+	Description string `json:"description"`
+	SendTo      string `json:"send_to"`
+	PostID      string `json:"post_id"`
+}
+
+func GetAddIssuePayloadFromJSON(data io.Reader) (*AddAPIRequest, error) {
+	body := &AddAPIRequest{}
+	if err := json.NewDecoder(data).Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func IsAddIssuePayloadValid(a *AddAPIRequest) error {
+	if a == nil {
+		return errors.New("invalid request body")
+	}
+
+	if a.Message == "" {
+		return errors.New("message is required")
+	}
+
+	return nil
+}
```

Comment:
Let's make this a method and simplify the name to `IsValid`
```suggestion
func (a *AddAPIRequest) IsValid error {
```

Reply:
@hanzei The reason why i was not able to do this because of the nill check as in the change you have suggested, We wont be able to access the functions of ```AddAPIRequest``` if it is already nill and will panic everytime.

Reply:
You can have a `nil` check in a method with a pointer receiver.